tensorflow/core/kernels/segment_reduction_ops.cc

```diff
@@ -269,13 +269,76 @@ class UnsortedSegmentSumOp : public OpKernel {
     }
   }
 };
+// Similar to SegmentReductionOp but can handle unsorted segment definitions and
+// specifying size of output.
+template <typename Device, class T, class Index>
+class UnsortedSegmentMaxOp : public OpKernel {
```

Comment:
Please consider refactoring the code here and in UnsortedSegmentSumOp into a base class UnsortedSegmentOpBase, appropriately templatized.